libs/configs.js

```diff
@@ -7,7 +7,7 @@ class configs {
         this.host = args.host;
         this.port = parseInt(args.port) || 8200;
         this.apiVersion = args.apiVersion || 'v1';
-        this.vaultLoginUrl = 'auth/aws/login';
+        this.vaultLoginUrl = args.vaultLoginUrl || 'auth/aws/login';
         this.vaultAppName = args.vaultAppName || process.env.AWS_LAMBDA_FUNCTION_NAME;
```

Comment:
it would be good if you encode the url too.

Reply:
I ran into a small issue adding the encoding. I made the assumption to use `encodeURI()`, but what I found is that it messes with the tests a bit, because encodeURI is trying to be extremely forgiving and returns a string value no matter what. (For example I pulled up my Chrome console, and did `encodeURI(undefined)` and the output is the string literal `"undefined"`, so doing something like:

`encodeURI(args.vaultLoginUrl) || 'auth/aws/login';` is no good, because it will actually never default.

I also tried getting slightly more fancy with:

`args.vaultLoginUrl ? encodeURI(vaultLoginUrl) ||  'auth/aws/login';`

But that also doesn't work with the existing expectations, because in this case, as long as the arg is defined at all (even as a boolean or a number), then we put it through `encodeURI()` which always produces a string, and therefore will always pass your validation.

That said, I can see a couple of ways to resolve that.

1) For the vaultLoginUrl, have the data type validation happen in the constructor. The down side here is that it decentralizes your validation, which makes testing uglier and creates an inconsistency with the overall validation flow.

OR (and I think this option flows better, but it's your library)

2) Wait to apply the encoding until the `getConfigs()` method. From what I noticed, it seems like `validateConfigs()` is aways called before `getConfigs()` so that way the validation is on the raw value, but can still remain in the same function as the rest of a validation, while also allowing that extra safety of encoding the login path before it's used.